arcgis-ios-sdk-samples/Content Display Logic/Controllers/Progress View Controller/ProgressViewController.swift

```diff
@@ -0,0 +1,156 @@
+    
+    private init() { }
+    
+    private var alertWindow: UIWindow?
+    
+    private var timer: Timer?
+    
+    func showProgress(message: String, duration: TimeInterval?, window: UIWindow?, animated: Bool) {
+        // Invalidate old timer, if one exists.
+        if let timer = timer {
+            timer.invalidate()
+            self.timer = nil
+        }
+        // Update existing progress if already presented.
+        if let alertWindow = alertWindow {
+            (alertWindow.rootViewController as! PhantomViewController).progressViewController.label.text = message
+        } else {
+            let progressWindow = UIWindow(frame: window?.frame ?? UIScreen.main.bounds)
+            let progressViewController = ProgressViewController()
+            progressViewController.loadViewIfNeeded()
+            progressViewController.label.text = message
+            let phantom = PhantomViewController(progressViewController: progressViewController)
+            progressWindow.rootViewController = phantom
+            progressWindow.windowLevel = .alert + 1
+            progressWindow.makeKeyAndVisible()
+            alertWindow = progressWindow
+        }
+        // Create a timer to dismiss if a duration is set.
+        if let duration = duration {
+            timer = Timer.scheduledTimer(withTimeInterval: duration, repeats: false) { [unowned self] _ in
+                self.timer = nil
+                self.hideProgress()
+            }
+        }
+    }
+    
+    func hideProgress() {
+        guard let window = alertWindow,
+              let phantom = window.rootViewController as? PhantomViewController
+        else { return }
+        
+        phantom.progressViewController.dismiss(animated: true) {
+            guard let alertWindow = self.alertWindow else { return }
+            alertWindow.resignKey()
+            self.alertWindow = nil
+        }
+    }
+}
+
+private class ProgressViewController: UIViewController {
+    @IBOutlet var label: UILabel!
+    @IBOutlet var activityView: UIActivityIndicatorView!
+    
+    init() {
+        super.init(nibName: "ProgressViewController", bundle: Bundle.main)
+        view.backgroundColor = backgroundColor(for: traitCollection.userInterfaceStyle)
+    }
+    
+    @available(*, unavailable)
+    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
+        fatalError("init(nibName:bundle:) has not been implemented")
+    }
+    
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        view.backgroundColor = backgroundColor(for: traitCollection.userInterfaceStyle)
+    }
+    
+    private func backgroundColor(for userInterfaceStyle: UIUserInterfaceStyle) -> UIColor {
```

Comment:
```suggestion
    func backgroundColor(for userInterfaceStyle: UIUserInterfaceStyle) -> UIColor {
```
Didn't see the private class. No need to private here.